### PR TITLE
use subprocess to call pip instead of the removed pip.main() interface

### PR DESCRIPTION
**Pull request by stgm**
**tl;dr** - fix for failing finance checks. My students will be using next week, so if at all possible, I would appreciate a deploy 😬

It seems that in the IDE at ide.legacy.cs50.io, the checks for Finance aren't working anymore. These checks depend on an install via a `requirements.txt`. Apparently the recent version of `pip` that is now available in the IDE is the root cause of this failure.

`check50` currently uses the `main()` method from the pip module. However, this has been removed a couple of versions ago, [according to this](https://stackoverflow.com/questions/12332975/installing-python-module-within-code/50255019#50255019).

In this PR, I've replaced that call with the use of the `subprocess` module, which was already imported anyway. I have tested this change in the IDE, with the Finance checks, and they once again work. I have not tested this change with any other checks.

Obv this needs to be deployed to the IDE to be of any use :-)

### CODE REVIEW

**Comment by stgm** [comment]
I see that the travis build on Python 3.4 is failing. I have used the 3.6 that is on the IDE to test this.

Apparently my * syntax is 3.5+ only, I'll let you decide if this is acceptable.

**Comment by crossroads1112** [comment]
Yep, this is fixed in v3 but not in v2 at present. Currently the way you've done it is going to leave some dead code, so I'm going to recommend replacing the following:

```
            try:
                code = subprocess.call([sys.executable, "-m", "pip", *args])
            except SystemExit as e:
                code = e.code

            if code:
                raise InternalError("failed to install dependencies in ({})".format(
                    requirements[len(config.args.checkdir) + 1:]))
```

with 

```
            try:
                code = subprocess.check_call([sys.executable, "-m", "pip", *args])
            except subprocess.CalledProcessError:
                raise InternalError("failed to install dependencies in ({})".format(
                    requirements[len(config.args.checkdir) + 1:]))
```

With that fixed, I'll approve.

**Comment by crossroads1112** [comment]
Regarding the failing checks, `check50` version 2 does need to work on python >= 2.7. v3 is 3.6 only, but until we roll that out, v2 needs to support python2. The suggested change above should remove the offending syntax

**Comment by stgm** [comment]
I have adopted your suggestions verbatim. Note that the exception handler is not actually triggered if there are unavailable packages in requirements.txt (I would argue that this is the most obvious use case for reporting an exception).

**Comment by stgm** [comment]
Ah, I think this was a misguided comment. The check contains a fresh requirements.txt which only contains the correct packages.

**Comment by kzidane** [comment]
Thank you @stgm and @crossroads1112!

**Comment by stgm** [comment]
Works like a charm. Nice!
